packages/block-library/src/navigation/edit/index.js

```diff
@@ -476,6 +477,36 @@ function Navigation( {
 		{ open: overlayMenuPreview }
 	);
 
+	if ( isLoading ) {
+		return (
+			<nav { ...blockProps }>
+				<Spinner className="wp-block-navigation__loading-indicator" />
+			</nav>
+		);
+	}
```

Comment:
I chatted just now to @tellthemachines about https://github.com/WordPress/gutenberg/issues/38169, and I'd be concerned this early return won't help matters because it'll cause `NavigationMenuSelector` to be unmounted when switching menus.

Reply:
Perhaps this would be an easy way out:

```js
let Loader;
if ( isLoading ) {
	Loader = (
```

and then below render it as `{ Loader }` ?

Reply:
Oh I see. Well we have a problem then because from a design perspective @jasmussen was hoping that the block would visually just show the spinner when "loading" and nothing else.

So perhaps we show the spinner but keep the block toolbar mounted?

Reply:
Showing only the spinner might make sense when a block is first added, or page first loads, but it feels a bit weird when the only thing we're doing is selecting a new menu to render in that place. I wouldn't even hide the toolbar in that case, I'd expect it to stay in place while the nav contents are switched up. After all, the only change is in the content, the block instance stays the same.

Reply:
Sorry for missing the ping here! One utopian version is we never show a spinner because things load so fast there's no need to indicate "processing". It's the same motivation that prompted the idea to defer the spinner to only appear if 2 seconds have passed.

Forgive me if I'm missing some technical aspects here, but if we are able to swap out the contents of a menu without showing a spinner (for instance by keeping the old menu present until the new one is able to take its place), that sounds great, especially as it would avoid layout shifts. Did I get that right?

Reply:
Ok I've changed the implementation so that the loading state isn't an early return. This means the spinner is rendered alongside the toolbars if they are available.

Note that this doesn't fix https://github.com/WordPress/gutenberg/issues/38169 because that appears to be a more complex problem but at least it won't make things worse. Thanks @tellthemachines for catching this one.

We'll need to add some e2e test coverage in any fix for https://github.com/WordPress/gutenberg/issues/38169 to ensure we don't have regressions.